lib/explorer/series.ex

```diff
@@ -841,6 +841,69 @@ defmodule Explorer.Series do
   def strftime(%Series{dtype: dtype}, _format_string),
     do: dtype_error("strftime/2", dtype, [:datetime])
 
+  @doc """
+  Clip (or clamp) the values in a series.
+
+  Values that fall outside of the interval defined by the min and max bounds
+  are clipped to the bounds.
+
+  ## Options
+
+  One of the bounds must be specified.
+
+    * `:min` - the minimum bound. Defaults to `nil` or no minimum clipping.
+    * `:max` - the maximum bound. Defaults to `nil` or no maximum clipping.
+
+  ## Supported dtypes
+
+    * `:integer`
+    * `:float`
+
+  Clipping other dtypes are possible using `select/3`.
+
+  ## Examples
+
+      iex> s = Explorer.Series.from_list([-50, 5, nil, 50])
+      iex> Explorer.Series.clip(s, min: 1, max: 10)
+      #Explorer.Series<
+        Polars[4]
+        integer [1, 5, nil, 10]
+      >
+
+      iex> s = Explorer.Series.from_list([-50, 5, nil, 50])
+      iex> Explorer.Series.clip(s, min: 1)
+      #Explorer.Series<
+        Polars[4]
+        integer [1, 5, nil, 50]
+      >
+
+      iex> s = Explorer.Series.from_list([-50, 5, nil, 50])
+      iex> Explorer.Series.clip(s, max: 10)
+      #Explorer.Series<
+        Polars[4]
+        integer [-50, 5, nil, 10]
+      >
+  """
+  @doc type: :element_wise
+  @spec clip(series :: Series.t(), opts :: Keyword.t()) :: Series.t()
+  def clip(_series, []) do
+    raise(
+      ArgumentError,
+      "Explorer.Series.clip/2 expects one of the minimum or the maximum " <>
+        "bounds to be specified"
+    )
+  end
+
+  def clip(%Series{dtype: dtype} = series, opts) when is_numeric_dtype(dtype) do
+    opts = Keyword.validate!(opts, min: nil, max: nil)
+    min = if is_integer(opts[:min]), do: opts[:min] / 1.0, else: opts[:min]
+    max = if is_integer(opts[:max]), do: opts[:max] / 1.0, else: opts[:max]
```

Comment:
I think we could check here if both of them is `nil`, and raise the same exception from the `clip/2` with empty options. This would prevent the `panic!` if the person passes `min: nil, max: nil` as options. WDYT?